cmake/macros/macro_enable_if_supported.cmake

```diff
@@ -43,21 +31,19 @@ MACRO(ENABLE_IF_SUPPORTED _variable _flag)
   # compilation unit.
   # Therefore we invert the test for -Wno-... flags:
   #
+  SET(_flag_sanitized "${_flag_stripped}")
   IF(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    STRING(REPLACE "-Wno-" "-W" _flag_stripped "${_flag_stripped}")
+    STRING(REPLACE "-Wno-" "-W" _flag_sanitized "${_flag_stripped}")
   ENDIF()
 
   IF(NOT "${_flag_stripped}" STREQUAL "")
     STRING(REGEX REPLACE "^-" "" _flag_name "${_flag_stripped}")
-    STRING(REPLACE "," "" _flag_name "${_flag_name}")
-    STRING(REPLACE "-" "_" _flag_name "${_flag_name}")
-    STRING(REPLACE "+" "_" _flag_name "${_flag_name}")
-    CHECK_CXX_COMPILER_FLAG(
-      "${_werror_string}${_flag_stripped}"
-      DEAL_II_HAVE_FLAG_${_flag_name}
-      )
+    STRING(REGEX REPLACE "\[-+,\]" "_" _flag_name "${_flag_name}")
+
+    CHECK_CXX_COMPILER_FLAG("${_flag_sanitized}" DEAL_II_HAVE_FLAG_${_flag_name})
```

Comment:
A related change in the `CHECK_CXX_COMPILER_FLAG` macro now requires a single compiler flag as argument. I.e., our old `-Werror <flag>` check no longer works because the flag is now handed down to the compiler as a single quoted argument.